fahrschein/src/main/java/org/zalando/fahrschein/ExponentialBackoffStrategy.java

```diff
@@ -81,4 +80,35 @@ public <T> T call(final int initialExceptionCount, final IOException initialExce
             }
         }
     }
+
+    @Override
+    public <T> T call(final int initialExceptionCount, final EventPersistenceException initialException,
```

Comment:
I attempted to consolidate the 'call' methods to reduce duplication, but it ended up making the code more complex, which, in my opinion, adversely affects readability. I'm open to suggestions.

```
private <T> T performRetry(final int initialExceptionCount, final IOException initialException,
            final Object callable) throws BackoffException, InterruptedException {
        checkMaxRetries(initialException, initialExceptionCount);
        int count = initialExceptionCount;
        if (count > 0) {
            sleepForRetries(count);
        }
        IOException lastRetryException = initialException;
        while (true) {
            try {
                LOG.trace("Try [{}]", count);
                if (callable instanceof ExceptionAwareCallable<?>) {
                    return ((ExceptionAwareCallable<T>) callable).call(count,
                            (EventPersistenceException) lastRetryException);
                }
                return ((IOCallable<T>) callable).call();
            } catch (EventPersistenceException e) {
                lastRetryException = e;
            } catch (IOException e) {
                lastRetryException = e;
                if (callable instanceof ExceptionAwareCallable<?>) {
                    throw new BackoffException(e, count);
                }
            } finally {
                LOG.warn("Got [{}] on retry [{}]", lastRetryException.getClass()
                        .getSimpleName(), count, lastRetryException);
                count++;
                checkMaxRetries(lastRetryException, count);
                sleepForRetries(count);
            }
        }
    }
```